tower_cli/models/base.py

```diff
@@ -423,6 +423,20 @@ def write(self, pk=None, create_on_missing=False, fail_on_found=False, force_on_
         # Actually perform the write.
         r = getattr(client, method.lower())(url, data=kwargs)
 
+        # Check for deprecated fields that failed to take effect
+        failed_fields = []
+        for field in self.fields:
+            if field.deprecated and kwargs.get(field.name) and field.name not in r:
+                # Uh oh, user using field that have been removed
+                failed_fields.append(field)
+        if failed_fields:
+            if method == 'POST':
+                debug.log('Due to field error, will attempt to delete created item.')
+                self.delete(r.json()['id'])
```

Comment:
I'm not sure I understand the intent here.  If they specified a deprecated field, delete the thing they just made?  This seems like fairly surprising behavior.

Reply:
> If they specified a deprecated field, delete the thing they just made?

There are 2 conditions that have to be met:

 - they specified a deprecated field
 - that deprecated field they specified failed to take effect

There is a very tense decision for the client to make - warning or error. Continuing without errors is going to cause havoc. But if the user sees an error, they would hope that no changes were committed. This is the closest we can get to some form of atomicity. Erring seems right to me. A warning would set the expectation that the object still exists, but would allow execution to continue. All options are messy, this is the best I could come up with.

Reply:
Yea, I don't think I like this behavior.  It's quite different from what, say, the `/api/` browser client does, and I think it's fairly surprising (maybe even more surprising than the field you specified not working).

Reply:
Maybe for fields like this that were deprecated which we removed, we should actually update the Tower API to detect when they're provided and raise an `HTTP 400` that explicitly describes that field no longer works.

Reply:
> It's quite different from what, say, the /api/ browser client does

It does what it does for REST principles, not usability. Therein lies the assumption that the client deals with problems like these (it has OPTIONS available, etc.), and this is the client we're talking about.

I would go even further in the direction of this PR, to wonder if maybe this ex post facto check should be universal in the client, for all fields it attempts to write. It's was likely to matter since tower-cli directly encodes the schema in code, but that changed with these fields.